In [2]:
import pandas as pd 
import datetime

In [3]:
# Chemin vers ton fichier CSV 
fichier = "C:/Users/Administrateur/OneDrive/Bureau/Projet 2 WCS/tmdb_full.csv"

# # Lire le fichier CSV 
df = pd.read_csv(fichier, sep =",") 

# Afficher le contenu du fichier CSV 
display(df.head())

C:\Users\Administrateur\AppData\Local\Temp\ipykernel_17108\2308964935.py:5: DtypeWarning: Columns (24) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(fichier, sep =",")


adult                     backdrop_path  budget  \
0  False  /dvQj1GBZAZirz1skEEZyWH2ZqQP.jpg       0   
1  False                               NaN       0   
2  False  /uJlc4aNPF3Y8yAqahJTKBwgwPVW.jpg       0   
3  False  /hQ4pYsIbP22TMXOUdSfC2mjWrO0.jpg       0   
4  False  /l94l89eMmFKh7na2a1u5q67VgNx.jpg       0   

                           genres homepage    id    imdb_id original_language  \
0                      ['Comedy']      NaN  3924  tt0029927                en   
1                   ['Adventure']      NaN  6124  tt0011436                de   
2            ['Drama', 'Romance']      NaN  8773  tt0055747                fr   
3    ['Drama', 'Comedy', 'Crime']      NaN     2  tt0094675                fi   
4  ['Drama', 'Comedy', 'Romance']      NaN     3  tt0092149                fi   

         original_title                                           overview  \
0               Blondie  Blondie and Dagwood are about to celebrate the...   
1   Der Mann ohne Namen                                                NaN   
2   L'Amour à vingt ans  Love at Twenty unites five directors from five...   
3                 Ariel  Taisto Kasurinen is a Finnish coal miner whose...   
4  Varjoja paratiisissa  An episode in the life of Nikander, a garbage ...   

   ...  runtime                spoken_languages    status  \
0  ...       70                          ['en']  Released   
1  ...      420                              []  Released   
2  ...      110  ['it', 'ja', 'pl', 'fr', 'de']  Released   
3  ...       73                          ['fi']  Released   
4  ...       76                          ['en']  Released   

                                             tagline  \
0  The favorite comic strip of millions at last o...   
1                                                NaN   
2               The Intimate Secrets of Young Lovers   
3                                                NaN   
4                                                NaN   

                           title  video vote_average vote_count  \
0                        Blondie  False        7.214          7   
1  Peter Voss, Thief of Millions  False        0.000          0   
2                 Love at Twenty  False        6.700         41   
3                          Ariel  False        7.046        248   
4            Shadows in Paradise  False        7.182        269   

                           production_companies_name  \
0                              ['Columbia Pictures']   
1                                                 []   
2  ['Ulysse Productions', 'Unitec Films', 'Cinese...   
3                      ['Villealfa Filmproductions']   
4                      ['Villealfa Filmproductions']   

             production_companies_country  
0                                  ['US']  
1                                      []  
2  ['', 'NZ', 'IT', 'JP', 'DE', 'PL', '']  
3                                  ['FI']  
4                                  ['FI']  

[5 rows x 25 columns]

In [4]:
df.columns

Index(['adult', 'backdrop_path', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_countries', 'release_date',
       'revenue', 'runtime', 'spoken_languages', 'status', 'tagline', 'title',
       'video', 'vote_average', 'vote_count', 'production_companies_name',
       'production_companies_country'],
      dtype='object')

In [5]:
df.shape

(309572, 25)

In [6]:
df['runtime'].describe()

count    309572.000000
mean         70.438702
std          95.136916
min           0.000000
25%          22.000000
50%          85.000000
75%          98.000000
max       43200.000000
Name: runtime, dtype: float64

In [7]:
df['status'].unique()

array(['Released', 'In Production', 'Post Production', 'Canceled',
       'Planned', 'Rumored'], dtype=object)

In [8]:
# Sélectionne les colonnes souhaitées
colonnes = ['genres', 'id', 'imdb_id', 'original_language', 
            'original_title', 'overview', 'poster_path', 'release_date', 
            'runtime', 'spoken_languages', 'status', 'title', 'video', 
            'vote_average', 'vote_count']

dfnew = df[colonnes]

# Afficher les 20 premières lignes du nouveau DataFrame
#print(dfnew.head(20))

dfnew.columns


Index(['genres', 'id', 'imdb_id', 'original_language', 'original_title',
       'overview', 'poster_path', 'release_date', 'runtime',
       'spoken_languages', 'status', 'title', 'video', 'vote_average',
       'vote_count'],
      dtype='object')

In [9]:
#filtrer les colonnes , visu valeurs manquantes , doublons
dfnew.value_counts().head(40)

genres                                                                        id       imdb_id     original_language  original_title                                                  overview                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               poster_path                       release_date  runtime  spoken_languages    status         title                          

In [10]:
# Je vais donc prendre que les films qui dure >= à 85min  ----> df['runtime'].describe()  50% : 85min 
condition = dfnew['runtime'] >= 85
dftime = dfnew[condition]

display(dftime.head(10))

genres    id    imdb_id  \
1                                ['Adventure']  6124  tt0011436   
2                         ['Drama', 'Romance']  8773  tt0055747   
5                          ['Crime', 'Comedy']     5  tt0113101   
6              ['Action', 'Crime', 'Thriller']     6  tt0107286   
9   ['Adventure', 'Action', 'Science Fiction']    11  tt0076759   
10                     ['Animation', 'Family']    12  tt0266543   
11              ['Comedy', 'Drama', 'Romance']    13  tt0109830   
12                                   ['Drama']    14  tt0169547   
13                        ['Mystery', 'Drama']    15  tt0033467   
14                          ['Drama', 'Crime']    16  tt0168629   

   original_language       original_title  \
1                 de  Der Mann ohne Namen   
2                 fr  L'Amour à vingt ans   
5                 en           Four Rooms   
6                 en       Judgment Night   
9                 en            Star Wars   
10                en         Finding Nemo   
11                en         Forrest Gump   
12                en      American Beauty   
13                en         Citizen Kane   
14                en   Dancer in the Dark   

                                             overview  \
1                                                 NaN   
2   Love at Twenty unites five directors from five...   
5   It's Ted the Bellhop's first night on the job....   
6   While racing to a boxing match, Frank, Mike, J...   
9   Princess Leia is captured and held hostage by ...   
10  Nemo, an adventurous young clownfish, is unexp...   
11  A man with a low IQ has accomplished great thi...   
12  Lester Burnham, a depressed suburban father in...   
13  Newspaper magnate, Charles Foster Kane is take...   
14  Selma, a Czech immigrant on the verge of blind...   

                         poster_path release_date  runtime  \
1   /6xUbUCvndklbGVYiljHr34NTxSl.jpg   1921-01-01      420   
2   /aup2QCYCsyEeQfpboXy0f4uj8aE.jpg   1962-06-22      110   
5   /75aHn1NOYXh4M7L5shoeQ6NGykP.jpg   1995-12-09       98   
6   /3rvvpS9YPM5HB2f4HYiNiJVtdam.jpg   1993-10-15      109   
9   /6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg   1977-05-25      121   
10  /5lc6nQc0VhWFYFbNv016xze8Jvy.jpg   2003-05-30      100   
11  /arw2vcBveWOVZr6pxd9XTd1TdQa.jpg   1994-06-23      142   
12  /wby9315QzVKdW9BonAefg8jGTTb.jpg   1999-09-15      122   
13  /sav0jxhqiH0bPr2vZFU0Kjt2nZL.jpg   1941-04-17      119   
14  /8Wdd3fQfbbQeoSfWpHrDfaFNhBU.jpg   2000-06-30      141   

                  spoken_languages    status                          title  \
1                               []  Released  Peter Voss, Thief of Millions   
2   ['it', 'ja', 'pl', 'fr', 'de']  Released                 Love at Twenty   
5                           ['en']  Released                     Four Rooms   
6                           ['en']  Released                 Judgment Night   
9                           ['en']  Released                      Star Wars   
10                          ['en']  Released                   Finding Nemo   
11                          ['en']  Released                   Forrest Gump   
12                          ['en']  Released                American Beauty   
13                          ['en']  Released                   Citizen Kane   
14                          ['en']  Released             Dancer in the Dark   

    video  vote_average  vote_count  
1   False         0.000           0  
2   False         6.700          41  
5   False         5.758        2357  
6   False         6.600         284  
9   False         8.208       18582  
10  False         7.823       17625  
11  False         8.481       24593  
12  False         8.027       10961  
13  False         8.021        4847  
14  False         7.881        1555

In [11]:
# Je selectionne fr et en dans original language
condition1 = dftime["original_language"].isin(["fr","en"])  #autre methode : condition = df["original_language"] == "fr" | df.original_language == "en"
df_en_fr = dftime[condition1]
print(df_en_fr.value_counts())

genres                                                                     id       imdb_id     original_language  original_title                     overview                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          

In [12]:
# verification 
df_en_fr['original_language'].unique()

array(['fr', 'en'], dtype=object)

In [13]:
#je selectionne les films release date sup à 1960 : release_date
df_en_fr['release_date'].info()

<class 'pandas.core.series.Series'>
Index: 82877 entries, 2 to 309570
Series name: release_date
Non-Null Count  Dtype 
--------------  ----- 
82115 non-null  object
dtypes: object(1)
memory usage: 1.3+ MB


In [14]:
#transformation en format datetime :
df_en_fr["release_date"]= pd.to_datetime(df_en_fr["release_date"])
print(df_en_fr.info())
print(df_en_fr['release_date'].head(2))

<class 'pandas.core.frame.DataFrame'>
Index: 82877 entries, 2 to 309570
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   genres             82877 non-null  object        
 1   id                 82877 non-null  int64         
 2   imdb_id            82877 non-null  object        
 3   original_language  82877 non-null  object        
 4   original_title     82877 non-null  object        
 5   overview           82090 non-null  object        
 6   poster_path        76465 non-null  object        
 7   release_date       82115 non-null  datetime64[ns]
 8   runtime            82877 non-null  int64         
 9   spoken_languages   82877 non-null  object        
 10  status             82877 non-null  object        
 11  title              82877 non-null  object        
 12  video              82877 non-null  bool          
 13  vote_average       82877 non-null  float64       
 14  vote_count

C:\Users\Administrateur\AppData\Local\Temp\ipykernel_17108\1019747106.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_en_fr["release_date"]= pd.to_datetime(df_en_fr["release_date"])


In [15]:
#condition 
condition2 = df_en_fr['release_date'] >= "1960-01-01"
dftime = df_en_fr[condition2]                        #dftime = df_en_fr[df_en_fr['release_date'] >= "1960-01-01"]
display(dftime.head(5))



genres    id    imdb_id  \
2                         ['Drama', 'Romance']  8773  tt0055747   
5                          ['Crime', 'Comedy']     5  tt0113101   
6              ['Action', 'Crime', 'Thriller']     6  tt0107286   
9   ['Adventure', 'Action', 'Science Fiction']    11  tt0076759   
10                     ['Animation', 'Family']    12  tt0266543   

   original_language       original_title  \
2                 fr  L'Amour à vingt ans   
5                 en           Four Rooms   
6                 en       Judgment Night   
9                 en            Star Wars   
10                en         Finding Nemo   

                                             overview  \
2   Love at Twenty unites five directors from five...   
5   It's Ted the Bellhop's first night on the job....   
6   While racing to a boxing match, Frank, Mike, J...   
9   Princess Leia is captured and held hostage by ...   
10  Nemo, an adventurous young clownfish, is unexp...   

                         poster_path release_date  runtime  \
2   /aup2QCYCsyEeQfpboXy0f4uj8aE.jpg   1962-06-22      110   
5   /75aHn1NOYXh4M7L5shoeQ6NGykP.jpg   1995-12-09       98   
6   /3rvvpS9YPM5HB2f4HYiNiJVtdam.jpg   1993-10-15      109   
9   /6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg   1977-05-25      121   
10  /5lc6nQc0VhWFYFbNv016xze8Jvy.jpg   2003-05-30      100   

                  spoken_languages    status           title  video  \
2   ['it', 'ja', 'pl', 'fr', 'de']  Released  Love at Twenty  False   
5                           ['en']  Released      Four Rooms  False   
6                           ['en']  Released  Judgment Night  False   
9                           ['en']  Released       Star Wars  False   
10                          ['en']  Released    Finding Nemo  False   

    vote_average  vote_count  
2          6.700          41  
5          5.758        2357  
6          6.600         284  
9          8.208       18582  
10         7.823       17625

In [16]:
#je selectionne les films deja realisés donc status : released
condition3 = dftime["status"].isin(["Released"])
dfreal = dftime[condition3]
display(dfreal.head())

genres    id    imdb_id  \
2                         ['Drama', 'Romance']  8773  tt0055747   
5                          ['Crime', 'Comedy']     5  tt0113101   
6              ['Action', 'Crime', 'Thriller']     6  tt0107286   
9   ['Adventure', 'Action', 'Science Fiction']    11  tt0076759   
10                     ['Animation', 'Family']    12  tt0266543   

   original_language       original_title  \
2                 fr  L'Amour à vingt ans   
5                 en           Four Rooms   
6                 en       Judgment Night   
9                 en            Star Wars   
10                en         Finding Nemo   

                                             overview  \
2   Love at Twenty unites five directors from five...   
5   It's Ted the Bellhop's first night on the job....   
6   While racing to a boxing match, Frank, Mike, J...   
9   Princess Leia is captured and held hostage by ...   
10  Nemo, an adventurous young clownfish, is unexp...   

                         poster_path release_date  runtime  \
2   /aup2QCYCsyEeQfpboXy0f4uj8aE.jpg   1962-06-22      110   
5   /75aHn1NOYXh4M7L5shoeQ6NGykP.jpg   1995-12-09       98   
6   /3rvvpS9YPM5HB2f4HYiNiJVtdam.jpg   1993-10-15      109   
9   /6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg   1977-05-25      121   
10  /5lc6nQc0VhWFYFbNv016xze8Jvy.jpg   2003-05-30      100   

                  spoken_languages    status           title  video  \
2   ['it', 'ja', 'pl', 'fr', 'de']  Released  Love at Twenty  False   
5                           ['en']  Released      Four Rooms  False   
6                           ['en']  Released  Judgment Night  False   
9                           ['en']  Released       Star Wars  False   
10                          ['en']  Released    Finding Nemo  False   

    vote_average  vote_count  
2          6.700          41  
5          5.758        2357  
6          6.600         284  
9          8.208       18582  
10         7.823       17625

In [17]:
dfreal.shape

(75756, 15)

In [18]:
# je selectionne les films ayant une note sup ou egale à 6/10 vote_average
condition4 = dfreal["vote_average"] >= 6
dfnote = dfreal[condition4]
display(dfnote.head())

genres    id    imdb_id  \
2                         ['Drama', 'Romance']  8773  tt0055747   
6              ['Action', 'Crime', 'Thriller']     6  tt0107286   
9   ['Adventure', 'Action', 'Science Fiction']    11  tt0076759   
10                     ['Animation', 'Family']    12  tt0266543   
11              ['Comedy', 'Drama', 'Romance']    13  tt0109830   

   original_language       original_title  \
2                 fr  L'Amour à vingt ans   
6                 en       Judgment Night   
9                 en            Star Wars   
10                en         Finding Nemo   
11                en         Forrest Gump   

                                             overview  \
2   Love at Twenty unites five directors from five...   
6   While racing to a boxing match, Frank, Mike, J...   
9   Princess Leia is captured and held hostage by ...   
10  Nemo, an adventurous young clownfish, is unexp...   
11  A man with a low IQ has accomplished great thi...   

                         poster_path release_date  runtime  \
2   /aup2QCYCsyEeQfpboXy0f4uj8aE.jpg   1962-06-22      110   
6   /3rvvpS9YPM5HB2f4HYiNiJVtdam.jpg   1993-10-15      109   
9   /6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg   1977-05-25      121   
10  /5lc6nQc0VhWFYFbNv016xze8Jvy.jpg   2003-05-30      100   
11  /arw2vcBveWOVZr6pxd9XTd1TdQa.jpg   1994-06-23      142   

                  spoken_languages    status           title  video  \
2   ['it', 'ja', 'pl', 'fr', 'de']  Released  Love at Twenty  False   
6                           ['en']  Released  Judgment Night  False   
9                           ['en']  Released       Star Wars  False   
10                          ['en']  Released    Finding Nemo  False   
11                          ['en']  Released    Forrest Gump  False   

    vote_average  vote_count  
2          6.700          41  
6          6.600         284  
9          8.208       18582  
10         7.823       17625  
11         8.481       24593

In [19]:
#verifier si doublons ? 
condition5 = dfnote.duplicated(['imdb_id'],keep=False)
doublons = dfnote[condition5]
print(doublons)

#pas de doublons

Empty DataFrame
Columns: [genres, id, imdb_id, original_language, original_title, overview, poster_path, release_date, runtime, spoken_languages, status, title, video, vote_average, vote_count]
Index: []


In [20]:
#verification des valeurs manquantes
dfnote.isnull().sum()

genres                  0
id                      0
imdb_id                 0
original_language       0
original_title          0
overview              217
poster_path          1336
release_date            0
runtime                 0
spoken_languages        0
status                  0
title                   0
video                   0
vote_average            0
vote_count              0
dtype: int64

In [21]:
dfnote.shape

(31447, 15)

In [22]:
#En vue du nombre de films et du nombre de données manquantes , je decide d'enlever les films n'ayant pas de valeur poster_path et pas d'overview
dfnote = dfnote.dropna()
display(dfnote.head())

genres    id    imdb_id  \
2                         ['Drama', 'Romance']  8773  tt0055747   
6              ['Action', 'Crime', 'Thriller']     6  tt0107286   
9   ['Adventure', 'Action', 'Science Fiction']    11  tt0076759   
10                     ['Animation', 'Family']    12  tt0266543   
11              ['Comedy', 'Drama', 'Romance']    13  tt0109830   

   original_language       original_title  \
2                 fr  L'Amour à vingt ans   
6                 en       Judgment Night   
9                 en            Star Wars   
10                en         Finding Nemo   
11                en         Forrest Gump   

                                             overview  \
2   Love at Twenty unites five directors from five...   
6   While racing to a boxing match, Frank, Mike, J...   
9   Princess Leia is captured and held hostage by ...   
10  Nemo, an adventurous young clownfish, is unexp...   
11  A man with a low IQ has accomplished great thi...   

                         poster_path release_date  runtime  \
2   /aup2QCYCsyEeQfpboXy0f4uj8aE.jpg   1962-06-22      110   
6   /3rvvpS9YPM5HB2f4HYiNiJVtdam.jpg   1993-10-15      109   
9   /6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg   1977-05-25      121   
10  /5lc6nQc0VhWFYFbNv016xze8Jvy.jpg   2003-05-30      100   
11  /arw2vcBveWOVZr6pxd9XTd1TdQa.jpg   1994-06-23      142   

                  spoken_languages    status           title  video  \
2   ['it', 'ja', 'pl', 'fr', 'de']  Released  Love at Twenty  False   
6                           ['en']  Released  Judgment Night  False   
9                           ['en']  Released       Star Wars  False   
10                          ['en']  Released    Finding Nemo  False   
11                          ['en']  Released    Forrest Gump  False   

    vote_average  vote_count  
2          6.700          41  
6          6.600         284  
9          8.208       18582  
10         7.823       17625  
11         8.481       24593

In [23]:
#verification / il ne devrait plus y avoir de valeurs manquantes 
dfnote.isnull().sum()

genres               0
id                   0
imdb_id              0
original_language    0
original_title       0
overview             0
poster_path          0
release_date         0
runtime              0
spoken_languages     0
status               0
title                0
video                0
vote_average         0
vote_count           0
dtype: int64

In [24]:
dfnote.shape

(29926, 15)

In [25]:
dfnote['vote_count'].describe()

count    29926.00000
mean       467.93768
std       1756.93224
min          0.00000
25%          4.00000
50%         16.00000
75%        123.00000
max      33630.00000
Name: vote_count, dtype: float64

In [26]:
#vu le nombre je vais filtrer en + sur le nombre de votes , avec nbres votes sup à 1000
condition6 = dfnote["vote_count"] >= 1000
dfnote = dfnote[condition6]
display(dfnote.head())

genres  id    imdb_id  \
9   ['Adventure', 'Action', 'Science Fiction']  11  tt0076759   
10                     ['Animation', 'Family']  12  tt0266543   
11              ['Comedy', 'Drama', 'Romance']  13  tt0109830   
12                                   ['Drama']  14  tt0169547   
14                          ['Drama', 'Crime']  16  tt0168629   

   original_language      original_title  \
9                 en           Star Wars   
10                en        Finding Nemo   
11                en        Forrest Gump   
12                en     American Beauty   
14                en  Dancer in the Dark   

                                             overview  \
9   Princess Leia is captured and held hostage by ...   
10  Nemo, an adventurous young clownfish, is unexp...   
11  A man with a low IQ has accomplished great thi...   
12  Lester Burnham, a depressed suburban father in...   
14  Selma, a Czech immigrant on the verge of blind...   

                         poster_path release_date  runtime spoken_languages  \
9   /6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg   1977-05-25      121           ['en']   
10  /5lc6nQc0VhWFYFbNv016xze8Jvy.jpg   2003-05-30      100           ['en']   
11  /arw2vcBveWOVZr6pxd9XTd1TdQa.jpg   1994-06-23      142           ['en']   
12  /wby9315QzVKdW9BonAefg8jGTTb.jpg   1999-09-15      122           ['en']   
14  /8Wdd3fQfbbQeoSfWpHrDfaFNhBU.jpg   2000-06-30      141           ['en']   

      status               title  video  vote_average  vote_count  
9   Released           Star Wars  False         8.208       18582  
10  Released        Finding Nemo  False         7.823       17625  
11  Released        Forrest Gump  False         8.481       24593  
12  Released     American Beauty  False         8.027       10961  
14  Released  Dancer in the Dark  False         7.881        1555

In [27]:
dfnote.shape

(2846, 15)

La plateforme Netflix possède 5000 films , on est ici à 2846 films 

In [28]:
df_final_tmdb = dfnote

In [30]:
# Transformer le DataFrame en fichier CSV
df_final_tmdb.to_csv('df_final_tmdb.csv', index=False)

# Message de confirmation
print("Le DataFrame a été sauvegardé en fichier CSV sous le nom 'df_final_tmdb.csv'.")


Le DataFrame a été sauvegardé en fichier CSV sous le nom 'df_final_tmdb.csv'.
